### PR TITLE
CP-23602: Add xapi API to toggle IGMP snooping

### PR DESCRIPTION
**Pull request by YarsinCitrix**
Updated according to review comments.

### CODE REVIEW

File: ocaml/idl/datamodel.ml [hunk 1]
```diff
@@ -7308,6 +7308,20 @@ let pool_disable_ssl_legacy = call
...
+    ~in_oss_since:None  
+    ~lifecycle:[  
+      Published, rel_inverness, "Set pool level IGMP Snooping enabled.";  
+    ]  
+    ~name:"set_igmp_snooping_enabled"  
+    ~params:[  
+      Ref _pool, "self", "The pool";  
+      Bool, "value", "Enable or disable IGMP Snooping on the pool"  
+    ]  
+    ~doc:"True to Enable IGMP Snooping in pool."
```
Comment (robhoes) [inline ocaml/idl/datamodel.ml]: I'd say "Enable or disable IGMP Snooping on the pool".

File: ocaml/xapi/nm.ml [hunk 2]
```diff
@@ -300,7 +302,9 @@ let rec create_bridges ~__context pif_rc net_rc =
     let ports = [pif_rc.API.pIF_device, {default_port with interfaces=[pif_rc.API.pIF_device]}] in
     cleanup,
     [net_rc.API.network_bridge, {default_bridge with ports; bridge_mac=(Some pif_rc.API.pIF_MAC);
-                                                     igmp_snooping = Some false; other_config; persistent_b=persistent}],
+                                                     igmp_snooping=(Some
```
Comment (robhoes) [inline ocaml/xapi/nm.ml]: Sorry to be pedantic, but what I meant earlier was:
```
let igmp_snooping = Db.Pool.get_igmp_snooping_enabled ~__context ~self:(Helpers.get_pool ~__context) in
cleanup,
[net_rc.API.network_bridge, {default_bridge with ports; bridge_mac=(Some pif_rc.API.pIF_MAC);
                                                 igmp_snooping; other_config; persistent_b=persistent}],
```

**Comment by huizh** [comment]
Hi @robhoes, is that OK for you to merge this PR? The implementation of the set API will be covered by another PR.

Thanks.

**Comment by huizh** [comment]
Thank you @robhoes

**Pull request merged**